salt/_pillar/metalk8s_nodes.py

```diff
@@ -92,9 +96,42 @@ def ext_pillar(minion_id, pillar, kubeconfig):
             for node in node_list.items
         )
 
+        volume_information = []
+
+        try:
+            object_response = customObjectsApi.list_cluster_custom_object(
+                group="storage.metalk8s.scality.com", version="v1alpha1",
+                plural="volumes"
+            )
+            if not object_response:
+                volume_information = __utils__['pillar_utils.errors_to_dict']([
+                    'Unable to retrieve node volume list'
+                ])
+            for items in object_response.get('items', []):
+                if items.get('spec', {}).get('nodeName') == minion_id:
```

Comment:
Here as well, no `nodeName` in the spec would be an invalid object, so just error out.